src/os/bluestore/AvlAllocator.h

```diff
@@ -246,7 +266,7 @@ class AvlAllocator : public Allocator {
 
   uint64_t _lowest_size_available() {
     auto rs = range_size_tree.begin();
-    return rs != range_size_tree.end() ? rs->length() : 0;
+    return rs != range_size_tree.end() ? rs->length() : std::numeric_limits<uint16_t>::max();
```

Comment:
uint16_t is obviously bad, taking into account that the intention was to mute condition:
`end - start > _lowest_size_available()`

Regardless, I think it is optimization at the cost of readability.
The function could have returned 0 - which nicely coworks with meaning "I have nothing",
and the caller would check it.

Reply:
Good catch! uint16_t is definitely wrong here.
But I'd prefer to keep max() return rather than 0. IMO it makes the caller's part shorter and more elegant.